server/chat-commands/moderation.ts

```diff
@@ -102,52 +100,56 @@ exports.commands = {
 			return this.errorReply(`Group 'room${Config.groups[nextGroup].id}' does not exist as a room rank.`);
 		}
 
-		let groupName = Config.groups[nextGroup].name || "regular user";
-		if ((room.auth[userid] || Config.groupsranking[0]) === nextGroup) {
+		const groupName = Config.groups[nextGroup].name || "regular user";
+		if (room.auth && (room.auth[userid] || Config.groupsranking[0]) === nextGroup) {
```

Comment:
You keep on unnecessarily checking for `room.auth` when like the second `if` in this function is `if (!room.auth) ...return...`

```suggestion
		if ((room.auth![userid] || Config.groupsranking[0]) === nextGroup) {
```